ui/dashboards/src/context/DatasourceStoreProvider.tsx

```diff
@@ -96,9 +99,18 @@ export function DatasourceStoreProvider(props: DatasourceStoreProviderProps) {
     async function getClient<Client>(selector: DatasourceSelector): Promise<Client> {
       const { kind } = selector;
       const [{ spec, proxyUrl }, plugin] = await Promise.all([findDatasource(selector), getPlugin('Datasource', kind)]);
-      return plugin.createClient(spec.plugin.spec, { proxyUrl }) as Client;
+
+      // allows extending client
+      const datasourceSpec = { ...spec.plugin.spec };
+      const client = plugin.createClient(datasourceSpec, { proxyUrl }) as Client;
+      if (onCreate !== undefined) {
+        if (isDatasourceClient(client)) {
+          return onCreate(client) as unknown as Client;
```

Comment:
Why the double `as`?

Reply:
Fixed this, I was having trouble with the types since DatasourceClient was essentially unknown before, but adding healthCheck and adjusting getClient helped